hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/management/JpaSoftwareModuleManagement.java

```diff
@@ -202,19 +205,23 @@ public Optional<SoftwareModule> get(final long id) {
     @Override
     public Optional<SoftwareModule> getByNameAndVersionAndType(final String name, final String version,
             final long typeId) {
+        assertSoftwareModuleTypeExists(typeId);
 
-        throwExceptionIfSoftwareModuleTypeDoesNotExist(typeId);
-
-        return softwareModuleRepository.findOneByNameAndVersionAndTypeId(name, version, typeId);
-    }
-
-    private boolean isUnassigned(final Long moduleId) {
-        return distributionSetRepository.countByModulesId(moduleId) <= 0;
+        // TODO AC : this method is used to validate input. Is it fine to restrict the
+        // access?
+        return JpaManagementHelper
+                .findOneBySpec(softwareModuleRepository, List.of(
+                        SoftwareModuleSpecification.likeNameAndVersion(name, version),
+                        SoftwareModuleSpecification.equalType(typeId),
+                        SoftwareModuleSpecification.fetchType()))
+                .map(SoftwareModule.class::cast);
     }
 
     private void deleteGridFsArtifacts(final JpaSoftwareModule swModule) {
+        // clearArtifactBinary validates unconditionally the module access
         for (final Artifact localArtifact : swModule.getArtifacts()) {
-            artifactManagement.clearArtifactBinary(localArtifact.getSha1Hash(), swModule.getId());
+            ((JpaArtifactManagement)artifactManagement)
+                    .clearArtifactBinary(localArtifact.getSha1Hash(), swModule.getId());
```

Comment:
Lets extend the ArtifactManagement interface by the `clearArtifactBinary` function to avoid the cast here.

Reply:
As wrote above I made the method even with package visibility. It not check access anymore so better be as much invisible as possible. 
Anyway no need to be public.

Reply:
I don't fully understand the reasoning. Method X checks whether the binaries can be deleted (if they are not used anywhere else). This is a valid check that does not take into account the access restrictions of the current user. In my opinion, nothing that needs to be hidden. An ArtifactManagement interface is intentionally autowired to allow multiple implementations. In case of a different implementation we have issues. In my opinion, the function required to delete artifacts should therefore also be part of the (public) interface.